pkg/apis/core/validation/validation_test.go

```diff
@@ -10517,6 +10517,7 @@ func TestValidateServiceCreate(t *testing.T) {
 			name: "valid load balancer protocol UDP 1",
 			tweakSvc: func(s *core.Service) {
 				s.Spec.Type = core.ServiceTypeLoadBalancer
+				s.Spec.AllocateLoadBalancerNodePorts = utilpointer.BoolPtr(true)
```

Comment:
Rather than doing this in every tweak, it should go in `makeValidService()`.  The tweaks are emphasizing what is DIFFERENT in each test case.

Reply:
The `AllocateLoadBalancerNodePorts` should only be set if the service type is `LoadBalancer`:
https://github.com/kubernetes/kubernetes/blob/7e23c5367dd6dc5b1eb304b395efebb1e01485a8/pkg/apis/core/types.go#L3733-L3737

And in the `makeValidService`, it's creating a service of type `ClusterIP`:
https://github.com/kubernetes/kubernetes/blob/4d38d21880a6f9c988fe5a1408ed71725f8084d8/pkg/apis/core/validation/validation_test.go#L9972-L9984

So for each `LoadBalancer` case, we have to do the change in the tweak function I think.